hack/bats/tests/preserve-env.bats

```diff
@@ -135,16 +125,93 @@ local_setup() {
     assert_line FOO=foo
     assert_line FOOBAR=foobar
     assert_line BAR=bar
-    refute_line --regexp '^BARBAZ='
+    assert_line BARBAZ=barbaz
```

Comment:
The test should add a variable from the default block list (e.g. `SSH_FOO`) to ALLOW to verify that it is being preserved. Add `FOO*` and `BAR` proves nothing because they would be preserved even if ALLOW is not set at all.

Furthermore this test should set another variable from the default block list, like `SSH_BAR` and validate that it is being blocked.